go-controller/pkg/crd/dnsobject/v1/types.go

```diff
@@ -0,0 +1,48 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +resource:path=dnsobject
+// +kubebuilder:resource:scope=Cluster
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="dnsObject Status",type=string,JSONPath=".status.status"
+// EgressFirewallDNS describes the current egressFirewallDNS entries for the cluster.
+// This is a centralized object to coordinate egressFirewallDNS entries. all nodes
+// update this object and the master node accesses the ovn database when needed.
+type DNSObject struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of EgressFirewallDNS
+	Spec DNSObjectSpec `json:"spec,omitempty"`
+
+	// Observed status of EgressFirewallDNS
+	// +optional
+	Status DNSObjectStatus `json:"status,omitempty"`
+}
+
+type DNSObjectSpec struct {
+	DNSObjectEntries map[string]DNSObjectEntry `json:"egressdnsEntries,omitempty"`
+}
+
+type DNSObjectEntry struct {
+	IPAddresses []string `json:"ipAddresses,omitempty"`
+}
+
+type DNSObjectStatus struct {
```

Comment:
Arguably, the map of name -> addresses belongs in Status.

Reply:
I had thought since that is the information that the object is relaying it wouldn't be a status. I plan on going back and adding status if the node is reporting errors with DNS